controller/src/beerocks/master/db/db.cpp

```diff
@@ -2418,6 +2479,431 @@ const std::list<sChannelScanResults> &db::get_channel_scan_results(const sMacAdd
+    // add entry to the persistent db
+    if (!bpl::db_add_entry(type_to_string(beerocks::eType::TYPE_CLIENT), db_entry, params)) {
+        LOG(ERROR) << "failed to add client entry " << db_entry << " to persistent db";
+        return false;
+    }
+    LOG(DEBUG) << "added client entry " << db_entry << " to persistent db";
+
+    return true;
+}
+
+std::chrono::steady_clock::time_point db::get_client_parameters_last_edit(const sMacAddr &mac)
+{
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found for mac " << mac;
+        return std::chrono::steady_clock::time_point::min();
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid client!";
+        return std::chrono::steady_clock::time_point::min();
+    }
+
+    return n->client_parameters_last_edit;
+}
+
+bool db::set_client_time_life_delay(const sMacAddr &mac,
+                                    const std::chrono::seconds &time_life_delay_sec,
+                                    bool save_to_persistent_db)
+{
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found for mac " << mac;
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid client!";
+        return false;
+    }
+
+    LOG(DEBUG) << __FUNCTION__ << ", time_life_delay_sec = " << time_life_delay_sec.count();
+
+    auto timestamp = std::chrono::steady_clock::now();
+    if (save_to_persistent_db) {
+        LOG(DEBUG) << __FUNCTION__
+                   << ", configuring persistent-db, timelife = " << time_life_delay_sec.count();
+
+        std::unordered_map<std::string, std::string> values_map;
+        values_map.at(timestamp_str)      = timestamp_to_string_seconds(timestamp);
+        values_map.at(timelife_delay_str) = std::to_string(time_life_delay_sec.count());
+
+        // update the persistent db
+        if (!update_client_entry_in_persistent_db(mac, values_map)) {
+            LOG(ERROR) << "failed to update client entry in persistent-db to for " << mac;
+            return false;
+        }
+    }
+
+    n->client_time_life_delay_sec  = time_life_delay_sec;
+    n->client_parameters_last_edit = timestamp;
+
+    return true;
+}
+
+std::chrono::seconds db::get_client_time_life_delay(const sMacAddr &mac)
+{
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found for mac " << mac;
+        return std::chrono::seconds::zero();
+    } else if (n->get_type() != beerocks::TYPE_CLIENT) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid client!";
+        return std::chrono::seconds::zero();
+    }
+
+    return n->client_time_life_delay_sec;
+}
+
+bool db::set_client_stay_on_initial_radio(const sMacAddr &mac, bool stay_on_initial_radio,
+                                          bool save_to_persistent_db)
+{
+    auto n = get_node(mac);
```

Comment:
```suggestion
    auto node = get_node(mac);
```

Reply:
replaced all new-code occurrences of `n` to `node`